src/no-enum.spec.ts

```diff
@@ -0,0 +1,156 @@
+// no-enum.spec.ts
+
+/*
+ * Copyright (c) 2021-2024 Check Digit, LLC
+ *
+ * This code is licensed under the MIT license (see LICENSE.txt for details).
+ */
+
+import rule, { ruleId } from './no-enum';
+import { RuleTester } from '@typescript-eslint/rule-tester';
+import { describe } from '@jest/globals';
+
+describe(ruleId, () => {
+  const ruleTester = new RuleTester();
+  ruleTester.run('no-side-effects', rule, {
+    valid: [
+      {
+        code: `const status = { SUCCESS: 'success', FAILURE: 'failure' }; type Status = 'success' | 'failure';`,
+      },
+      {
+        code: `const shapes = { CIRCLE: 'circle', SQUARE: 'square', TRIANGLE: 'triangle' }; type Shape = 'circle' | 'square' | 'triangle';`,
+      },
+    ],
+    invalid: [
+      {
+        code: `enum Status { SUCCESS = 'success', FAILURE = 'failure' };`,
+        errors: [
+          {
+            messageId: 'NO_ENUM',
+          },
+        ],
+      },
+      {
+        code: `enum Days { MONDAY = 'Monday', TUESDAY = 'Tuesday', WEDNESDAY = 'Wednesday' };`,
+        errors: [
+          {
+            messageId: 'NO_ENUM',
+          },
+        ],
+      },
+      {
+        code: `export const test = {
+                properties: {
+                  testString1: {
+                    type: 'string'
+                  },
+                  testType: {
+                    type: 'string',
+                    enum: ['TEST_KEY'],
```

Comment:
i'm not sure if we want to prohibit `enum` in JSON schema definition. maybe @carlansley can comment on this?